### PR TITLE
client: SAST (SonarLint) Fixes and update CWEs

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOG > Add maint note.
- ExtensionClientIntegration > Use lambdas vs inner classes where possible. Use constants from InputEvent vs. KeyEvent.
- MissingUrlsthread > Make method static where practical.
- OptionsPassiveScan > Remove useless semi-colon. Simplify stream toList handling.
- PopupMenuClientOpenInBrowser > Use lambdas vs inner classes where possible.
- PopupMenuItemClient > Use method reference vs lambda.
- PopupMenuItemClientDetails > Use method reference vs lambda.
- PopupMenuItemClientHistory > Use method reference vs lambda.
- PopupMenuItemClientOpenInBrowser > Remove unnecessary curly braces.
- PopupMenuSpider > Return statement directly vs using if where practical. Streamline switch cases with comma notation.
- SensitiveInfoInStorageScanRule > Adjust enum values to use full caps, and have a name to be used for resources messages. Adjust regex to use \d instead of [0-9]. Make methods static where practical.
- Messages.properties > Make KVPs match enum values.
- Scan rules & unit tests > Updated CWEs to not use 200.

## Related issues
- zaproxy/zaproxy#8712

## Checklist
- [na] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [na] Write tests
- [na] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by thc202** [comment]
The build is failing.

**Comment by psiinon** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [69a00720\-7e5c\-4a56\-8834\-2899d2e2855e](https://eu.ast.checkmarx.net/projects/0da811fa-6373-42b1-bbb6-d0026285b136/scans?id=69a00720-7e5c-4a56-8834-2899d2e2855e&branch=kingthorin%2Fclient-fix)
### No New Or Fixed Issues Found

**Comment by kingthorin** [comment]
Now also includes scan rules changes to address usage of CWE-200.

**Review by psiinon — APPROVED** [review]

File: addOns/client/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## Unreleased
 ### Changed
 - Update minimum ZAP version to 2.16.0.
+- Maintenance changes.
+- The current passive scan rules now uses a more specific CWE (Issue 8732).
```
Comment (thc202) [inline addOns/client/CHANGELOG.md:10]: The issue is not correct.

File: addOns/client/src/main/resources/org/zaproxy/addon/client/resources/Messages.properties [hunk 2]
```diff
@@ -66,12 +66,12 @@ client.pscan.jwtinstorage.stdname = Information Disclosure - JWT in Browser Stor
...
+client.pscan.seninfoinstorage.other.EMAIL = The following data (key=value) was set which matches the pattern for email addresses: {0}\nNote that alerts will only be raised once for each URL + key.
+client.pscan.seninfoinstorage.other.SSN = The following data (key=value) was set which matches the pattern for US social security numbers: {0}\nNote that alerts will only be raised once for each URL + key.
+client.pscan.seninfoinstorage.other.base64.CC = The following data (key=value) was set which matches the pattern for credit cards: {0}\nThe value can be base64 decoded: {1}\nNote that alerts will only be raised once for each URL + key.
+client.pscan.seninfoinstorage.other.base64.EMAIL = The following data (key=value) was set which matches the pattern for email addresses: {0}\nThe value can be base64 decoded: {1}\nNote that alerts will only be raised once for each URL + key.
+client.pscan.seninfoinstorage.other.base64.SSN = The following data (key=value) was set which matches the pattern for US social security numbers: {0}\nThe value can be base64 decoded: {1}\nNote that alerts will only be raised once for each URL + key.
```
Comment (thc202) [inline addOns/client/src/main/resources/org/zaproxy/addon/client/resources/Messages.properties:74]: Remove the dependency on the enum name instead.

Comment (thc202) [reply addOns/client/src/main/resources/org/zaproxy/addon/client/resources/Messages.properties]: Key or I18n, name is misleading.

**Comment by kingthorin** [comment]
Hopefully got those.

**Comment by kingthorin** [comment]
Tweaked to getId

File: addOns/client/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## Unreleased
 ### Changed
 - Update minimum ZAP version to 2.16.0.
+- Maintenance changes.
+- The current passive scan rules now uses a more specific CWE (Issue 8712).
```
Comment (thc202) [inline addOns/client/CHANGELOG.md:10]: `use`

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**